sql/dd/impl/bootstrap/bootstrap_ctx.h

```diff
@@ -123,10 +124,16 @@ class DD_bootstrap_ctx {
     m_actual_I_S_version = actual_I_S_version;
   }
 
+  void set_actual_dd_engine(uint actual_dd_engine) {
+    m_actual_dd_engine = actual_dd_engine;
```

Comment:
This can be called at most once, consider adding `assert(m_actual_dd_engine == DB_TYPE_UNKNOWN);`